specs/BadgeNotifications/BadgeNotifications-spec.md

```diff
@@ -0,0 +1,222 @@
+# Badge Notifications in Windows App SDK
+
+This is the spec for proposal: [Issue #138](https://github.com/microsoft/WindowsAppSDK/issues/138)
+
+# Background
+
+A notification badge conveys summary or status information specific to your app. They can be numeric
+(1-99) or one of a set of system-provided glyphs. Examples of information best conveyed through a
+badge include network connection status in an online game, user status in a messaging app, number of
+unread mails in a mail app, and number of new posts in a social media app.
+
+Example of a numeric badge in Taskbar (One new message in Whatsapp):
+
+![Screenshot](NumericBadgeExample.png)
+
+Example of a system-provided glyph in taskbar (New message in Outlook):
+
+![Screenshot](GlyphBadgeExample.png)
+
+This Badge notification functionality is missing in WinAppSDK, which not only blocks apps like
+WhatsApp to move to WinAppSDK but also stops them from availing new features and goodness that we
+are building in WinAppSDK. The scope of this document is to fill that gap and provide badge
+notification capability in WinAppSDK as well.
+
+# Description
+
+Badge notifications provide users with a visual indicator of new or important content in your
+application. This feature enhances user engagement by allowing them to quickly see updates without
+needing to open the app. Badge notifications are already supported for UWP apps. This initiative is
+to make badge notifications work in WindowsAppSdk.
+
+For more details see:
+
+-   [Badge Notification WinRT APIs](https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/badges)
+    Defines all the API constructs that we have for Badge Notifications in WinRT today.
+
+# Compatibility
+### Supported Platforms
+This API is exclusively available for packaged apps. It is not supported for web apps or other 
+types of applications.
+### Notification Type
+The badge notifications are local to the device. No remote server interaction is required or 
+supported for updating the badge count.
+
+
+# Examples
+
+## Create a numeric badge (c++)
+
+```c++
+private void setBadgeNumber(uint32_t num)
+{
+   winrt::BadgeNotification badgeNotification(num);
+   winrt::BadgeNotificationManager manager = winrt::BadgeNotificationManager::Current();
+   manager.Update(badgeNotification);
```

Comment:
The general rule is that [if a method is a verb without a direct object, then the implied direct object is the object whose method is being called](https://devblogs.microsoft.com/oldnewthing/20240724-00/?p=110037).

So `manager.Update()` is updating the manager. `manager.Clear()` is clearing the manager.

Should these be
* manager.UpdateBadge(badgeNotification)
* manager.ClearBadge()

?

Reply:
updated